# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1468593658 14400
#      Fri Jul 15 10:40:58 2016 -0400
# Node ID 03e30a5d6e0d8f6669caa164b6a531755f387662
# Parent  b017351e92e1e744defe194eef27bf4a8a77861f
skip explicit instantiations of xmin and xmax if they are not defined

* MArray-i.cc, MArray-s.cc: Only instantiate xmin and xmax if
OCTAVE_USE_DEPRECATED_FUNCTIONS is defined.

diff --git a/liboctave/array/MArray-i.cc b/liboctave/array/MArray-i.cc
--- a/liboctave/array/MArray-i.cc
+++ b/liboctave/array/MArray-i.cc
@@ -31,22 +31,24 @@ along with Octave; see the file COPYING.
 #include "MArray.h"
 #include "MArray.cc"
 
 template class OCTAVE_API MArray<int>;
 #if defined (OCTAVE_ENABLE_64)
 template class OCTAVE_API MArray<int64_t>;
 #endif
 
+#if defined (OCTAVE_USE_DEPRECATED_FUNCTIONS)
 // Explicit instantiation, as this seems to be required by weird compilers
 // like MSVC.  This should be harmless on other compilers.
 template int xmin<int> (int, int);
 template int xmax<int> (int, int);
 template long xmin<long> (long, long);
 template long xmax<long> (long, long);
+#endif
 
 INSTANTIATE_MARRAY_FRIENDS (int, OCTAVE_API)
 #if defined (OCTAVE_ENABLE_64)
 INSTANTIATE_MARRAY_FRIENDS (int64_t, OCTAVE_API)
 #endif
 
 template class OCTAVE_API MArray<octave_int8>;
 template class OCTAVE_API MArray<octave_int16>;
diff --git a/liboctave/array/MArray-s.cc b/liboctave/array/MArray-s.cc
--- a/liboctave/array/MArray-s.cc
+++ b/liboctave/array/MArray-s.cc
@@ -26,20 +26,22 @@ along with Octave; see the file COPYING.
 
 // Instantiate MArrays of short int values.
 
 #include "MArray.h"
 #include "MArray.cc"
 
 template class OCTAVE_API MArray<short>;
 
+#if defined (OCTAVE_USE_DEPRECATED_FUNCTIONS)
 // Explicit instantiation, as this seems to be required by weird compilers
 // like MSVC.  This should be harmless on other compilers.
 template short xmin<short> (short, short);
 template short xmax<short> (short, short);
+#endif
 
 INSTANTIATE_MARRAY_FRIENDS (short, OCTAVE_API)
 
 #include "MDiagArray2.h"
 #include "MDiagArray2.cc"
 
 template class OCTAVE_API MDiagArray2<short>;
 
